detalizers/callStack.go

```diff
@@ -0,0 +1,98 @@
+//   Copyright © 2015-2017 Ivan Kostko (github.com/ivan-kostko; github.com/gopot)
+
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+
+//       http://www.apache.org/licenses/LICENSE-2.0
+
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package detalizers
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+	"sync"
+)
+
+const (
+	CallStackDetailKey = callStackKey("Call Stack")
+)
+
+// Represents type alias for Call Stack key
+type callStackKey string
+
+// Implements fmt.Stringer() which is consumed by DetailedError/Error interfaces
+// to generate Detailed() result.
+func (this callStackKey) String() string {
+	return string(this)
+}
+
+// Represents internal implementation of Call Stack value. Should implement fmt.Stringer
+type callStackValue struct {
+	frames  *runtime.Frames
+	once    sync.Once
+	printed string
+}
+
+// Represents errors.Detalizer implementation to generate call stack detail!
```

Comment:
Could you add a small comment what the parameters to this function do? It is not clear without reading the code

Reply:
Valid point